.github/workflows/api-binary-compatibility.yml

```diff
@@ -27,6 +27,9 @@ on:
       - 'apache-iceberg-**'
   pull_request:
     paths:
+      - '.github/**'
```

Comment:
my assumption is that .github/ isn't changed often, so we don't have to optimize this, and it would be a miss not to trigger on chages to eg shared composite actions (.github/actions/....) -- we don't have such yet, but i'd assume we will have such sooner or later.

Reply:
done!